### PR TITLE
[primTorch] add exp2 (prim and ref), log10 (prim and ref), frac (ref)

### PR DESCRIPTION
**Pull request by kshitij12345**
Adds `exp2`, `log10` to the prims (both also exist in C++ lib and Intel SIMD intrinsic has `exp2`)

Adds `exp2`, `log10`, `frac` to refs with corresponding entries to OpInfo.

Tried to decompose `exp2` (before adding it as prim) as
* `exp(log(2) * x)` but it wasn't stable at large numbers.
* `pow(2, x)` in which case there was stride mismatch. At cursory look, `pow` tries to preserve stride of first arg if possible.

Tried to decompose `log10` (before adding it as prim) as
* `log(x) / log(10)` passed for real dtypes. Failed for complex at extremals. Probably related to https://github.com/pytorch/pytorch/issues/52332 (not a 100% sure)

### CODE REVIEW

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -355,12 +358,31 @@ def _ref(a: TensorLikeType) -> TensorLikeType:
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT,
 )
 
+exp2 = _make_elementwise_unary_reference(
+    prims.exp2,
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT,
+    aten_op=torch.ops.aten.exp2,
```
Comment (mruberry) [inline torch/_refs/__init__.py:364]: This shouldn't be necessary because its name is already `exp2`, right?

File: torch/_refs/__init__.py [hunk 2]
```diff
@@ -355,12 +358,31 @@ def _ref(a: TensorLikeType) -> TensorLikeType:
...
+def _frac(x: TensorLikeType) -> TensorLikeType:
+    # frac(x) = x - floor(abs(x)) * sign(x)
+    trunc_x = mul(floor(abs(x)), sign(x))
+    return sub(x, trunc_x)
+
+
+frac = _make_elementwise_unary_reference(
+    _frac,
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
+    aten_op=torch.ops.aten.frac,
```
Comment (mruberry) [inline torch/_refs/__init__.py:382]: Same here -- I think we can get away without this?

File: torch/_refs/__init__.py [hunk 3]
```diff
@@ -355,12 +358,31 @@ def _ref(a: TensorLikeType) -> TensorLikeType:
...
+)
+
 floor = _make_elementwise_unary_reference(
     prims.floor,
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
 )
 
 
+def _frac(x: TensorLikeType) -> TensorLikeType:
+    # frac(x) = x - floor(abs(x)) * sign(x)
```
Comment (mruberry) [inline torch/_refs/__init__.py:374]: TODO: delete me

**Review by mruberry — APPROVED** [review]
Really nice work, @kshitij12345!

Just a few small inline cleanup notes.

File: torch/_refs/__init__.py (hunk 1 above)
Comment (kshitij12345) [reply torch/_refs/__init__.py:364]: Ah, right. Had added it while prototyping the decomposition. Will remove it. Thanks!

File: torch/_refs/__init__.py (hunk 2 above)
Comment (kshitij12345) [reply torch/_refs/__init__.py:382]: I think this is required, since `aten_op` won't be inferred with `_frac`.

Comment (mruberry) [reply torch/_refs/__init__.py:382]: Oh you're totally right -- sorry, I didn't notice the underscore

**Pull request closed without merging**